homeassistant/components/jewish_calendar/__init__.py

```diff
@@ -5,104 +5,155 @@
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-def get_unique_prefix(
-    location: Location,
-    language: str,
-    candle_lighting_offset: int | None,
-    havdalah_offset: int | None,
-) -> str:
-    """Create a prefix for unique ids."""
-    config_properties = [
-        location.latitude,
-        location.longitude,
-        location.timezone,
-        location.altitude,
-        location.diaspora,
-        language,
-        candle_lighting_offset,
-        havdalah_offset,
-    ]
-    prefix = "_".join(map(str, config_properties))
-    return f"{prefix}"
-
-
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Jewish Calendar component."""
     if DOMAIN not in config:
         return True
 
-    name = config[DOMAIN][CONF_NAME]
-    language = config[DOMAIN][CONF_LANGUAGE]
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+        )
+    )
+
+    return True
 
-    latitude = config[DOMAIN].get(CONF_LATITUDE, hass.config.latitude)
-    longitude = config[DOMAIN].get(CONF_LONGITUDE, hass.config.longitude)
-    diaspora = config[DOMAIN][CONF_DIASPORA]
 
-    candle_lighting_offset = config[DOMAIN][CONF_CANDLE_LIGHT_MINUTES]
-    havdalah_offset = config[DOMAIN][CONF_HAVDALAH_OFFSET_MINUTES]
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up a configuration entry for Jewish calendar."""
+    name = config_entry.data.get(CONF_NAME, DEFAULT_NAME)
+    language = config_entry.data.get(CONF_LANGUAGE, DEFAULT_LANGUAGE)
+    diaspora = config_entry.data.get(CONF_DIASPORA, DEFAULT_DIASPORA)
+
+    if not config_entry.options:
+        # If options are not defined, update entry with optional values from the
+        # original configuration.
+        options = {
+            CONF_CANDLE_LIGHT_MINUTES: config_entry.data.get(
+                CONF_CANDLE_LIGHT_MINUTES, DEFAULT_CANDLE_LIGHT
+            ),
+            CONF_HAVDALAH_OFFSET_MINUTES: config_entry.data.get(
+                CONF_HAVDALAH_OFFSET_MINUTES, DEFAULT_HAVDALAH_OFFSET_MINUTES
+            ),
+        }
+
+        hass.config_entries.async_update_entry(config_entry, options=options)
+
+    # The current specification for config_entries of jewish_calendar allows to
+    # optionally specify latitude and longitude. So we can have a config_entry with
+    # CONF_LAT/LON specified, one with CONF_LOCATION or None in which case we'll
+    # take the Home location.
+    if CONF_LATITUDE in config_entry.data or CONF_LONGITUDE in config_entry.data:
+        latitude = config_entry.data.get(CONF_LATITUDE, hass.config.latitude)
+        longitude = config_entry.data.get(CONF_LONGITUDE, hass.config.longitude)
+    elif CONF_LOCATION in config_entry.data:
+        latitude = config_entry.data[CONF_LOCATION][CONF_LATITUDE]
+        longitude = config_entry.data[CONF_LOCATION][CONF_LONGITUDE]
```

Comment:
What? We can have lat long in entry data and lat long in location entry data? Why do we do this?